File: blog-backend/blog-pojo/src/main/java/com/yvmoux/blog/entity/Article.java
package com.yvmoux.blog.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("article")
public class Article {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;

    private String summary;

    private String aiSummary;

    private String coverImage;

    private Long authorId;

    private String status;

    private Long viewCount;

    private Long likeCount;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: blog-backend-go/internal/model/article.go
package model

import "time"

// Article maps to the `article` table.
type Article struct {
	ID           int64     `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	Title        string    `gorm:"column:title;size:200;not null" json:"title"`
	Summary      string    `gorm:"column:summary;size:500" json:"summary"`
	AiSummary    string    `gorm:"column:ai_summary;type:longtext" json:"aiSummary"`
	AiSummaryLong string   `gorm:"column:ai_summary_long;type:longtext" json:"aiSummaryLong"`
	CoverImage   string    `gorm:"column:cover_image;size:255" json:"coverImage"`
	AuthorID     int64     `gorm:"column:author_id;not null;index:idx_author" json:"authorId"`
	Status       string    `gorm:"column:status;size:20;not null;default:PUBLISHED" json:"status"`
	ViewCount    int64     `gorm:"column:view_count;not null;default:0" json:"viewCount"`
	LikeCount    int64     `gorm:"column:like_count;not null;default:0" json:"likeCount"`
	CreatedAt    time.Time `gorm:"column:created_at;not null;index:idx_created" json:"createdAt"`
	UpdatedAt    time.Time `gorm:"column:updated_at;not null" json:"updatedAt"`
}

func (Article) TableName() string {
	return "article"
}
